janusgraph-test/src/test/java/org/janusgraph/graphdb/query/QueryTest.java

```diff
@@ -131,6 +132,45 @@ public void testOrderList() {
 
     }
 
+    @Test
+    public void testIndexQueryCache() {
```

Comment:
These tests do not cover the case in which we have a limit `x` first and then a limit `y > x` later. Otherwise the tests look good

Reply:
Agree. I think I will need to break this test case into multiple smaller test cases. Now several scenarios are mixed together and edge cases are more likely to be missing.